src/earn/EarnEnterAmount.tsx

```diff
@@ -202,30 +228,45 @@ function EarnEnterAmount({ route }: Props) {
         localAmount: parsedLocalAmount,
       }
     }
-  }, [tokenAmountInput, localAmountInput, enteredIn, token])
+  }, [tokenAmountInput, localAmountInput, enteredIn, transactionToken, mode])
 
-  const feeCurrencies = useSelector((state) => feeCurrenciesSelector(state, token.networkId))
+  const feeCurrencies = useSelector((state) =>
+    feeCurrenciesSelector(state, transactionToken.networkId)
+  )
+
+  const balanceInTransactionToken = useMemo(
+    () =>
+      isWithdrawal
+        ? transactionToken.balance.multipliedBy(pool.pricePerShare[0])
+        : transactionToken.balance,
+    [transactionToken, pool, mode]
+  )
 
   useEffect(() => {
     clearPreparedTransactions()
 
     if (
       !tokenAmount ||
       tokenAmount.isLessThanOrEqualTo(0) ||
-      tokenAmount.isGreaterThan(token.balance)
+      tokenAmount.isLessThanOrEqualTo(0) ||
+      tokenAmount.isGreaterThan(balanceInTransactionToken)
     ) {
       return
     }
     const debouncedRefreshTransactions = setTimeout(() => {
-      return handleRefreshPreparedTransactions(tokenAmount, token, feeCurrencies)
+      return handleRefreshPreparedTransactions(
+        isWithdrawal ? tokenAmount.multipliedBy(pool.pricePerShare[0]) : tokenAmount,
```

Comment:
this conversion looks wrong

Reply:
Fixed in 4af772d15b12693af80d1b3a0193c532d32bfa78!